lib/matplotlib/cbook.py

```diff
@@ -1657,7 +1657,11 @@ def safe_isfinite(val):
         raise RuntimeError("matplotlib does not "
                            "support generators as input")
     else:
-        return next(val for val in obj if safe_isfinite(val))
+        try:
+            return next(val for val in obj if safe_isfinite(val))
+        except TypeError:
+            # not an iterable...
+            return obj
```

Comment:
Is this still necessary? It looks like the calls to `_safe_first_finite` were reverted.

Reply:
Thats true... OTOH, if we pass `_safe_first_element(12)` do we want it to error?